### PR TITLE
Update WCSLIB to version 8.1

### PR DESCRIPTION
**Pull request by mcara**
### Description
This PR updates built-in WCSLIB distributed with `astropy`. Excerpts from WCSLIB's changelog:

```
WCSLIB version 8.1 (2023/07/06)
-------------------------------

Changes in the contents of the auxprm struct (in 8.0 beta) necessitated
incrementing the major version number from 7 to 8, i.e. the ABI changed.
Took the opportunity for minor tweaks to wcsprm and disprm.

* C library

  - In disprm, changed the order of maxdis and totdis to ensure correct
    alignment of doubles on 32-bit machines.

WCSLIB version 8.0 beta (2023/07/01)
------------------------------------

Beta test version, not publically released.

* C library

  - Support planetary keywords A_RADIUS, B_RADIUS, C_RADIUS, BLON_OBS,
    BLAT_OBS, and BDIS_OBS in auxprm by analogy with the Solar keywords
    added at 7.1.  Requested by Chiara Marmo (Observatoire de Paris).

  - Added wcsprm::time to the wcsprm struct to record the TIME axis
    number (along with lng, lat, and spec).

  - Fixed a bug in wcspih() and wcsbth() where trailing blanks were not
    stripped from string keyvalues.  Reported by Naveen Dukiya.

  - Fixed a problem affecting thread safety in disp2x(), disx2p(), and
    diswarp().  Reported by Mohammad Akhlaghi.

```

For a full list of changes, see [WCSLIB's CHANGELOG](http://www.atnf.csiro.au/people/mcalabre/WCS/CHANGES)
or [astropy/cextern/wcslib/CHANGES](https://github.com/astropy/astropy/blob/9fb2ea735cbc1865d28c970a4e1d98e93794e829/cextern/wcslib/CHANGES)

This update is to support work on https://github.com/astropy/astropy/pull/14820.

Many thanks to @cmarmo !

### CODE REVIEW

**Review by nden — APPROVED** [review]
LGTM

File: docs/changes/wcs/15035.feature.rst
Comment (pllim) [inline docs/changes/wcs/15035.feature.rst:1]: We typically don't backport feature. So I would suggest `other` if you think this really should go into 5.3.x. Also, Simon is already doing 5.3.1 so this cannot get in until 5.3.2 the earliest.

**Review by pllim — CHANGES_REQUESTED** [review]

**Review by pllim — APPROVED** [review]
Mihai and I discussed the milestone offline. We agreed now that this does not need to be backported, so it will land in v6.0. Thanks, all!

**Comment by pllim** [comment]
@cmarmo , you can pick this up now if you rebase. 😸

**Comment by pllim** [comment]
FYI, I had to fix your change log in d59d7fa211a3cb81c34f52d85f55b6e9ac0d136c because RTD was failing with a warning:

```
docs/changelog.rst:5: WARNING: Duplicate explicit target name: "astropy/cextern/wcslib/changes".
```

**Pull request merged**